Makefile

```diff
@@ -0,0 +1,149 @@
+TRIVY_ADAPTER_DOWNLOAD_URL=https://github.com/aquasecurity/harbor-scanner-trivy/releases/download/$(TRIVYADAPTERVERSION)/harbor-scanner-trivy_$(TRIVYADAPTERVERSION:v%=%)_Linux_x86_64.tar.gz
+
+
+SEDCMD=$(shell which sed)
+SEDCMDI=$(SEDCMD) -i
+ifeq ($(shell uname),Darwin)
+    SEDCMDI=$(shell which gsed) -i
+endif
+
+# dockerhub user
+REGISTRYUSER=
+REGISTRYPASSWORD=
+
+_update_makefile:
+	@echo "update goharbor makefile"
+	@$(SEDCMDI) 's/--rm/--rm --env CGO_ENABLED=0 --env GOOS=linux --env GOARCH=arm64/g' $(HARBOR_MAKEFILE_PATH);
+	@$(SEDCMDI) 's/$$(DOCKERBUILD)/docker buildx build --platform linux\/arm64 --progress plain --output=type=docker/g' $(HARBOR_MAKEFILE_PATH)
+
+_update_make_photon_makefile:
+	@echo "update goharbor photon makefile"
+	@$(SEDCMDI) 's/$(DOCKERCMD) build/$(DOCKERCMD) buildx build --platform linux\/arm64 --progress plain --output=type=docker/' $(HARBOR_PHOTON_MAKEFILE_PATH)
+
+_update_chartserver:
+	@echo "update goharbor chartserver compile.sh"
+	@$(SEDCMDI) 's/go build -a/GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -a/g' $(HARBOR_PHOTON_CHARTSERVER_COMPILE_PATH)
+
+_update_registry:
+	@echo "update goharbor Dockerfile.binary"
+	@$(SEDCMDI) 's/CGO_ENABLED=0/GOOS=linux GOARCH=arm64 CGO_ENABLED=0/g' $(HARBOR_PHOTON_REGISTRY_DOCKERFILE_PATH)
+
+_update_trivy-adapter:
+	@echo "update goharbor trivy-adapter Dockerfile.binary"
+	@$(SEDCMDI) 's/CGO_ENABLED=0/GOARCH=arm64 CGO_ENABLED=0/g' $(HARBOR_PHOTON_TRIVY-ADAPTER_DOCKERFILE_PATH)
+
+_update_portal:
+	@echo "update goharbor portal Dockerfile"
+	@$(SEDCMDI) 's/node:15.4.0/--platform=$${BUILDPLATFORM:-linux\/amd64} node:15.4.0/g' $(HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH);
+	@$(SEDCMDI) "s/'node_modules\/@angular\/cli\/bin\/ng'/.\/node_modules\/@angular\/cli\/bin\/ng/g" $(HARBOR_PHOTON_PORTAL_DOCKERFILE_PATH)
+
+_update_notary:
+	@echo "update goharbor notary binary.Dockerfile"
+	@$(SEDCMDI) '8 a ENV CGO_ENABLED 0 \nENV GOOS linux \nENV GOARCH arm64' $(HARBOR_PHOTON_NOTARY_DOCKERFILE_PATH)
+
+_update_pushimage:
+	@echo "update pushimage.sh"
+	@$(SEDCMDI) 's/grep "goharbor"/grep "harborarmdev"/g' $(HARBOR_PUSHIMAGE_PATH)
+
+pre_update: _update_makefile _update_make_photon_makefile _update_chartserver _update_registry _update_trivy-adapter _update_notary _update_portal
+
+# downlaod goharbor/harbor source code
+download:
+	$(shell git clone --branch $(HARBOR_TAG) $(HARBOR_SOURCE_URL) $(SRCPATH))
+	@echo "download goharbor/harbor source code success"
+
+compile: download pre_update
+	cd $(SRCPATH) && make -f Makefile compile
+
+build_base_image: download pre_update
+	cd $(SRCPATH) && make -f Makefile build_base_docker -e BUILD_PG96=$(BUILD_PG96) -e PUSHBASEIMAGE=yes
+
+.PHONY: build
+build: compile
+	cd $(SRCPATH) && make -f $(MAKEFILEPATH_PHOTON)/Makefile $(BUILDTARGET) -e DEVFLAG=$(DEVFLAG) -e GOBUILDIMAGE=$(GOBUILDIMAGE) \
+	 -e REGISTRYVERSION=$(REGISTRYVERSION) -e REGISTRY_SRC_TAG=$(REGISTRY_SRC_TAG) \
```

Comment:
do not need to pass these versions, just let goharbor to use the default ones.

Reply:
Some dependencies need to be replaced with the arm version link, so the dependency link needs to be rewritten, like TRIVY_DOWNLOAD_URL, TRIVY_ADAPTER_DOWNLOAD_URL